discovery_engine/lib/src/ffi/types/trending_topic_vec.dart

```diff
@@ -0,0 +1,71 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import 'dart:ffi' show Pointer;
+
+import 'package:xayn_discovery_engine/src/domain/models/trending_topic.dart';
+import 'package:xayn_discovery_engine/src/ffi/genesis.ffigen.dart';
+import 'package:xayn_discovery_engine/src/ffi/load_lib.dart' show ffi;
+import 'package:xayn_discovery_engine/src/ffi/types/list.dart';
+import 'package:xayn_discovery_engine/src/ffi/types/trending_topic.dart';
+
+final _adapter =
+    ListFfiAdapter<TrendingTopicFfi, RustTrendingTopic, RustVecTrendingTopic>(
+  alloc: ffi.alloc_uninitialized_trending_topic_slice,
+  next: ffi.next_trending_topic,
+  writeNative: (trendingTopic, place) => trendingTopic.writeNative(place),
+  readNative: (place) => TrendingTopicFfi.readNative(place),
```

Comment:
:art: the function could be passed directly as the closure
```suggestion
  readNative: TrendingTopicFfi.readNative,
```